godot-core/src/obj/raw.rs

```diff
@@ -0,0 +1,624 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::ptr;
+
+use godot_ffi as sys;
+use sys::{interface_fn, GodotFfi, GodotNullableFfi, PtrcallType};
+
+use crate::builtin::meta::{
+    ClassName, FromGodot, GodotCompatible, GodotFfiVariant, GodotType, ToGodot,
+};
+use crate::builtin::{Variant, VariantConversionError};
+use crate::obj::dom::Domain as _;
+use crate::obj::mem::Memory as _;
+use crate::obj::{dom, mem, GodotClass};
+use crate::obj::{GdMut, GdRef, InstanceId};
+use crate::storage::InstanceStorage;
+use crate::{engine, out};
+
+/// Low-level bindings for object pointers in Godot.
+///
+/// This should not be used directly, you should either use [`Gd<T>`](super::Gd) or [`Option<Gd<T>>`]
+/// depending on whether you need a nullable object pointer or not.
+#[repr(C)]
+pub struct RawGd<T: GodotClass> {
```

Comment:
Can we not hide it completely from the API with `#[doc(hidden)]`?

(For now, I appreciate being able to check the docs online though😉)